docs/rfcs/0002_templates_mvp.md

```diff
@@ -0,0 +1,339 @@
+
+## ✨ Creating Templates
+
+### CreateTemplateMutation
+
+Creates a Template and initial TemplateVersion
+
+```graphql
+type CreateTemplateInput {
+    name: String!
+    owner: ID!
+    description: String!
+    fieldsSchema: JSONString
+    fieldsSchemaFile: Upload
+}
+```
+
+### CreateTemplateVersionMutation
+
+Creates a TemplateVersion and adds it to a Template
+
+```graphql
+type CreateTemplateVersionInput {
+    templateID: ID!
+    description: String!
+    fieldsSchema: JSONString
+    fieldsSchemaFile: Upload
+}
+```
+
+### (Optional) CloneTemplateMutation
+
+Creates a new Template and a TemplateVersion by duplicating the
+referenced template version and it's template. Overwrites the name,
+description, and owner of the source Template with the supplied values
+in the cloned Template.
+
+```graphql
+type CloneTemplateInput {
+    templateVersion: ID!
+    name: String!
+    description: String!
+    owner: ID
+}
+```
+
+### Permissions
+
+-   Any organization or study admin may create or clone Templates/TemplateVersions
+
+## 🛠 Modifying Templates
+
+### UpdateTemplateMutation
+
+```graphql
+type UpdateTemplateInput {
+    name: String
+    description: String
+    owner: ID
+}
+```
+
+### UpdateTemplateVersionMutation
+
+```graphql
+type UpdateTemplateVersionInput {
+    description: String
+    fieldsSchema: JSONString
+    fieldsSchemaFile: Upload
+}
+```
+
+### Permissions
+
+-   Any organization or study administrator may update a Template or
+    TemplateVersion if their organization owns it
+
+## 🗑 Deleting Templates
+
+### DeleteTemplateMutation
+
+```graphql
+type DeleteTemplateInput {
+    id: ID!
+}
+```
+
+### DeleteTemplateVersionMutation
+
+```graphql
+type DeleteTemplateVersionInput {
+    id: ID!
+}
+```
+
+### Permissions
+
+### Permissions
+
+-   Any organization or study administrator may delete a Template or
+    TemplateVersion if their organization owns it
+
+## 📦 Downloading Templates
+
+### Download Study Templates
+
+Probably the easiest way to download templates since it bundles everything
+into an archive. You will get an archive containing:
+
+For each TemplateVersion in the study:
+
+-   TemplateVersion.submission_template_file: the TSV file that people will fill out and use for data submission to their study
+-   TemplateVersion.fields_schema: the template field definitions which serve as instructions on filling out the template
+
+    /download/study/<Study.kf_id>templates
+
+We should consider providing a multi-sheet Excel workbook with a sheet for each
+template's fields schema and a folder of TSVs with all of the submission
+templates.
+
+### Download Latest Template
+
+Download the latest version of template.
+
+```
+/download/template/<Template.id>
+```
+
+You would get an archive with two files:
+
+-   Template.latest_version.submission_template_file
+-   Template.latest_version.fields_schema
```

Comment:
See earlier comment about `fields_schema`. I think that's the most important part of the "template" because it defines each field, provides type, instructions, etc for the actual submission template